tests/tests_integration/test_configs.py

```diff
@@ -77,6 +81,90 @@ def test_bad_paths(self, project, bad_pattern, path_type):
 
+    @pytest.mark.parametrize("path_type", ["local_path", "central_path"])
+    def test_non_existant_local_path(
+        self, no_cfg_project, tmp_path, path_type
+    ):
+        """
+        Check that if the `local_path` and `central_path` that holds the
+        project root does not exist when passed to configs, that an error
+        is raised. Note that this error is only raised for `central_path`
+        if `connection_method` is `"local_filesystem"`.
+        See `test_additional_error_text_when_ssh_used()` for when ssh is used.
+        """
+        (
+            non_existant_path,
+            existant_path,
+        ) = self.get_exists_and_does_not_exist_path(tmp_path)
+
+        if path_type == "local_path":
+            local_path = non_existant_path
+            central_path = existant_path
+        else:
+            local_path = existant_path
+            central_path = non_existant_path
+
+        with pytest.raises(BaseException) as e:
+            no_cfg_project.make_config_file(
+                local_path / no_cfg_project.project_name,
+                central_path / no_cfg_project.project_name,
+                "local_filesystem",
+            )
+
+        assert f"The {path_type}: {non_existant_path} that the project" in str(
+            e.value
+        )
+
+    def test_additional_error_text_when_ssh_used(
+        self, no_cfg_project, tmp_path
+    ):
+        """
+        If SSH is used as `connection_method`, if `local_path` does not exist
+        an extra message is printed to warn to check the `central_path`, because
+        it cannot be checked.
+
+        Currently if SSH is used and the central path does not exist,
+        no error is raised because it is not possible to check.
+        """
+        (
+            non_existant_path,
+            existant_path,
+        ) = self.get_exists_and_does_not_exist_path(tmp_path)
+
+        with pytest.raises(BaseException) as e:
+            no_cfg_project.make_config_file(
+                non_existant_path / no_cfg_project.project_name,
+                existant_path / no_cfg_project.project_name,
+                "ssh",
+                central_host_id="fake_id",
+                central_host_username="fake_username",
+            )
+
+        assert (
+            "Also make sure the central_path`, is correct, as datashuttle "
+            "cannot check it via SSH at this stage." in str(e.value)
+        )
+
+        # This should not raise an error, even though the path does not
+        # exist, because it is not possible to check over SSH.
+        no_cfg_project.make_config_file(
+            existant_path / no_cfg_project.project_name,
+            non_existant_path / no_cfg_project.project_name,
+            "ssh",
+            central_host_id="fake_id",
+            central_host_username="fake_username",
+        )
+
+    def get_exists_and_does_not_exist_path(self, tmp_path):
```

Comment:
Nitpicking here, but I wonder whtehr it's worth defining these two paths as fixtures. Alternatively, keep this functon as is but return a dict instead of tuple.

Reply:
Also, I think the spelling should be "existent" (instead of "existant")

Reply:
Thanks good idea! I think I originally did not do this because it needs `tmp_path` which is also  fixture; forgetting that fixtures can take fixtures as arguments. Fixed [here](https://github.com/neuroinformatics-unit/datashuttle/pull/286/commits/47da8d0eee9f4be390b4fee901f325e6d30bac66).